### PR TITLE
Push Docker image

### PR DESCRIPTION
**Pull request by giorgiosironi**
Needs to be on an origin branch rather than a fork to access environment variables for Docker Hub access.

Should normally only be executed on a tag or on `master`.

Without a Docker Hub organization it's too early to differentiate between stable and unstable, so this will be an image with a certain tag that will be deployed to the `unstable` environment as an additional step when it becomes available.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,13 @@ language: minimal
 
 services: docker
 
+branches:
+    only:
+        - master
```
Comment (giorgiosironi) [inline .travis.yml]: better way to instrument Travis CI than the point and click interface, allows the mainline and PRs to be built but not all pushed branches

File: .docker/push-image.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+# should be executed only if $TRAVIS_BRANCH = master
```
Comment (giorgiosironi) [inline .docker/push-image.sh]: continuing tomorrow

File: .docker/push-image.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (thewilkybarkid) [inline .docker/push-image.sh]: Might be better to put `.travis`.

File: .travis.yml [hunk 4]
```diff
@@ -17,3 +24,4 @@ script:
 
 after_script:
     - docker-compose -f docker-compose.yaml down
+    - .docker/push-image.sh
```
Comment (thewilkybarkid) [inline .travis.yml]: Will push after failures too. Probably ok.

Comment (giorgiosironi) [reply .travis.yml]: That's _really_ unstable, will move to `after_success`

File: .travis.yml [hunk 5]
```diff
@@ -17,3 +24,6 @@ script:
 
 after_script:
     - docker-compose -f docker-compose.yaml down
+
+after_success:
```
Comment (thewilkybarkid) [inline .travis.yml]: Looks like it's [run before `after_script`](https://docs.travis-ci.com/user/customizing-the-build/#the-build-lifecycle), would be useful to keep the same order for clarity.

File: .travis/push-image.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+set -e
+
+if [[ "$TRAVIS_BRANCH" = "master" && "$TRAVIS_PULL_REQUEST" = "false" ]]; then
+    echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USERNAME" --password-stdin 
+
+    # tag temporarily as liberoadmin due to lack of `libero/` availability
+    docker tag "libero/browser:$IMAGE_TAG" "liberoadmin/browser:$IMAGE_TAG"
+    docker push "liberoadmin/browser:$IMAGE_TAG"
+else
+    echo "not pushing an image, we are not on master"
```
Comment (thewilkybarkid) [inline .travis/push-image.sh]: Fine for now. We can improve messages when we update to use the proper organisation and handle tags.

**Comment by giorgiosironi** [comment]
First run in https://travis-ci.com/libero/browser/builds/82548926

**Pull request merged**